src/main/java/Sam.java

```diff
@@ -0,0 +1,203 @@
+                    }
+                    StringBuilder listOutput = new StringBuilder("Here are the tasks in your list:\n");
+                    for (int i = 0; i < tasks.size(); i++) {
+                        listOutput.append((i + 1)).append(". ").append(tasks.get(i)).append("\n");
+                    }
+                    return listOutput.toString().trim();
+
+                case MARK: {
+                    int idx = parseIndex(rest, tasks.size());
+                    tasks.get(idx).markDone();
+                    storage.save(tasks.getTasks());
+                    return "Nice! I've marked this task as done:\n" + tasks.get(idx);
+                }
+
+                case UNMARK: {
+                    int idx = parseIndex(rest, tasks.size());
+                    tasks.get(idx).unmark();
+                    storage.save(tasks.getTasks());
+                    return "OK, I've marked this task as not done yet:\n" + tasks.get(idx);
+                }
+
+                case DELETE: {
+                    int idx = parseIndex(rest, tasks.size());
+                    Task removed = tasks.remove(idx);
+                    storage.save(tasks.getTasks());
+                    return "Noted. I've removed this task:\n" + removed + 
+                           "\nNow you have " + tasks.size() + " tasks in the list.";
+                }
+
+                case TODO: {
+                    if (rest.isEmpty()) {
+                        throw new EmptyDescriptionException("todo");
+                    }
+                    tasks.add(new Todo(rest));
+                    storage.save(tasks.getTasks());
+                    return "Got it. I've added this task:\n" + tasks.get(tasks.size() - 1) + 
+                           "\nNow you have " + tasks.size() + " tasks in the list.";
+                }
+
+                case DEADLINE: {
+                    if (rest.isEmpty() || !rest.contains("/by")) {
+                        throw new SamException("OOPS!!! Use: deadline <description> /by <time>");
+                    }
+                    String[] a = rest.split("/by", 2);
+                    String descr = a[0].trim(), by = a[1].trim();
+                    if (descr.isEmpty() || by.isEmpty()) {
+                        throw new SamException("OOPS!!! Use: deadline <description> /by <time>");
+                    }
+                    tasks.add(new Deadline(descr, by));
+                    storage.save(tasks.getTasks());
+                    return "Got it. I've added this task:\n" + tasks.get(tasks.size() - 1) + 
+                           "\nNow you have " + tasks.size() + " tasks in the list.";
+                }
+
+                case EVENT: {
+                    if (rest.isEmpty() || !rest.contains("/from") || !rest.contains("/to")) {
+                        throw new SamException("OOPS!!! Use: event <description> /from <start> /to <end>");
+                    }
+                    String[] p1 = rest.split("/from", 2);
```

Comment:
Would it be clearer to rename p1 to startDate? It might make the purpose of the variable more immediately obvious to readers, especially in contexts involving date handling.
```suggestion
                    String[] startDate = rest.split("/from", 2);
```